ansible/roles/daliuge_arl/tasks/daliuge.yml

```diff
@@ -0,0 +1,99 @@
+---
+# daliuge_arl/tasks/daliuge.yml
+#
+# NOTE: Defaults are loaded from daliuge_arl/defaults/main.yml
+#
+# Update and/or install pip packages
+- pip:
+    name: pip setuptools
+    state: latest
+
+# Remove any old copy of DALiuGE
+- file:
+    path: "{{ VIRTENV_LOC }}"
+    state: absent
+
+## Setup virtualenv -- COULDN'T GET THIS TO WORK
+##- name: Initiate virtualenv
+#  pip: name=dal_arl_venv
+#       virtualenv=dal_arl_venv
+#       extra_args="-m venv"
+#       virtualenv_command=python36
+#       virtualenv_python=python3.6
```

Comment:
We've tackled using virtualenvs before using this pattern: https://github.com/stackhpc/ansible-role-os-images/blob/master/tasks/main.yml#L129-L230 - would that help here, possibly in the role's top-level task file?